app/delivery/send_to_providers.py

```diff
@@ -141,22 +141,29 @@ def check_for_malware_errors(document_download_response_code, notification):
     return error codes if the scan is in progress or if malware was detected.
     This function contains the logic for handling these errors.
     """
-    # 423 "Locked" response is sent if malicious content was detected
-    if document_download_response_code == 423:
-        current_app.logger.info(
-            f"Malicious content detected! Download and attachment failed for notification.id: {notification.id}"
-        )
-        # Update notification that it contains malware
-        malware_failure(notification=notification)
-
-    # 428 "Precondition Required" response is sent if the scan is still in progress
-    if document_download_response_code == 428:
-        current_app.logger.info(f"Malware scan in progress, could not download files for notification.id: {notification.id}")
-        # Throw error so celery will retry in sixty seconds
-        raise MalwareScanInProgressException
 
-    if document_download_response_code != 200:
-        document_download_internal_error(notification=notification)
+    match document_download_response_code:
+        # 423 "Locked" response is sent if malicious content was detected
+        case 423:
+            current_app.logger.info(
+                f"Malicious content detected! Download and attachment failed for notification.id: {notification.id}"
+            )
+            # Update notification that it contains malware
+            malware_failure(notification=notification)
+        # 428 "Precondition Required" response is sent if the scan is still in progress
+        case 428:
+            current_app.logger.info(f"Malware scan in progress, could not download files for notification.id: {notification.id}")
+            # Throw error so celery will retry
+            raise MalwareScanInProgressException
+        # 408 "Request Timeout" response is sent if the scan does is not complete before it times out
+        case 408:
+            current_app.logger.info(f"Malware scan timed out for notification.id: {notification.id}, send anyway")
```

Comment:
We might already have it, but can we have an alarm (in a TF PR) to tell us when scan-files times out?

Reply:
CC @andrewleith over here